lib/pat.c

```diff
@@ -2905,29 +2905,32 @@ grn_pat_nextid(grn_ctx *ctx, grn_pat *pat, const void *key, uint32_t key_size)
 }
 
 static void
-get_tc(grn_ctx *ctx, grn_pat *pat, grn_hash *h, pat_node_common *rn)
+get_transitive_closure(grn_ctx *ctx,
+                       grn_pat *pat,
+                       grn_hash *hash,
+                       pat_node_common *root_node)
```

Comment:
How about renaming `root_node` to `current_node` (or similar)? After the initial call, it actually refers to the node currently being processed in the recursion, not the true root node.

```suggestion
                       pat_node_common *current_node)
```